### PR TITLE
Merge 5.6.27 2

### PR DESCRIPTION
**Pull request by vlad-lesin**
Merge tag 'mysql-5.6.27' into merge-5.6.27-2 …

Added testcases for:
74589
74778
74851
74950
75230
76946
75667

As new warnings about wrong utf8 symbols take place the results are fixed for:
ctype_utf8
ctype_uca
ctype_uca_innodb
# The following commit:

commit e5cb1d5
Author: Thirunarayanan Balathandayuthapani thirunarayanan.balathandayuth@oracle.com
Date:   Wed Jul 8 12:40:32 2015 +0530

```
Bug #19779113   INNODB REFUSES STARTUP WITH INNODB_FORCE_RECOVERY>3
```
# 

is reverted because it breaks the old logic where it is supposed
INNODB_FORCE_RECOVERY>3 is read only mode but not all places are fixed
according to the new logic.

For example if srv_force_recovery==SRV_FORCE_NO_LOG_REDO then srv_read_only_mode
must be set, otherwise redo log can be updated
(for example from trx_sys_flush_max_trx_id()).

As SAVEPOINT replication issue is solved in .27, the correspondent Percona fixes are reverted. The same is true for "Memory leak in add_derived_key()", innodb_buffer_pool_populate and mysqld_safe numa_interleave options.

http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/1006/

### CODE REVIEW

**Comment by laurynas-biveinis** [comment]
I am not sure about NUMA change integration.
Removing options in the middle of GA series will make users angry. For innodb_buffer_pool_populate, it is possible to make it an alias for the upstream innodb_numa_interleave one (achieves the same effect), and for mysqld_safe.sh, if --numa-interleave is given, better to pass --innodb-numa-interleave to mysqld?

File: mysql-test/t/bug-74589.test [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Bug 74589. A severe error occurred during the log apply phase of an online
```
Comment (laurynas-biveinis) [inline mysql-test/t/bug-74589.test]: --source include/have_innodb.inc
suite/innodb

File: mysql-test/t/bug-74851.test [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Bug 74851. With foreign_key_checks disabled, an assertion was raised when
+# adding a foreign key constraint on a renamed column that was part of
+# a previous foreign key constraint.
```
Comment (laurynas-biveinis) [inline mysql-test/t/bug-74851.test]: --source include/have_innodb.inc

Comment (laurynas-biveinis) [inline mysql-test/t/bug-74851.test]: InnoDB suite

File: mysql-test/t/bug-75230.test [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Bug 75230.
+# INNODB: CRASH WHEN CONVERTING A TABLE WITH
+# UTF8 COLUMN TO ROW_FORMAT=REDUNDAN
```
Comment (laurynas-biveinis) [inline mysql-test/t/bug-75230.test]: --source include/have_innodb.inc, InnoDB suite

File: mysql-test/t/bug-75667.test [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Bug 75667. In certain cases, ALTER TABLE ... REBUILD PARTITION was not
+# handled correctly when executed on a locked table.
```
Comment (laurynas-biveinis) [inline mysql-test/t/bug-75667.test]: --source include/have_partition.inc, parts suite

**Comment by laurynas-biveinis** [comment]
What is the status of 75677/77667 ?

**Comment by laurynas-biveinis** [comment]
t/percona_bug1308016.test is a strict subset of innodb.innodb-lock-inherit-read_commited. Usually we leave old testcases around, but here it's essentially a duplicate, maybe let's remove our version?

**Comment by vlad-lesin** [comment]
Could you please review once more? http://jenkins.percona.com/job/percona-server-5.6-param/1007/

**Pull request merged**